config/sources/families/rk35xx.conf

```diff
@@ -1,42 +1,42 @@
 source "${BASH_SOURCE%/*}/include/rockchip64_common.inc"
 
-BOOTSOURCE='https://github.com/radxa/u-boot.git'
-BOOTBRANCH='branch:stable-4.19-rock3'
-BOOTPATCHDIR="u-boot-rk35xx"
-OVERLAY_PREFIX='rk35xx'
-
-if [[ "${BOOT_SOC}" == rk3566 ]]; then
-	CPUMAX=1800000
-	DDR_BLOB='rk35/rk3566_ddr_1056MHz_v1.10.bin'
-	BL31_BLOB='rk35/rk3568_bl31_v1.29.elf'
-elif [[ "${BOOT_SOC}" == rk3568 ]]; then
-	CPUMAX=1992000
-	DDR_BLOB='rk35/rk3568_ddr_1056MHz_v1.10.bin'
-	BL31_BLOB='rk35/rk3568_bl31_v1.29.elf'
```

Comment:
That's right, they should be in the family, but the family is for everyone RK - rockchip64.  :)
I think all the common settings (frequency setting, etc.) should be in one place, it makes no sense to duplicate them in different places. Moreover, it will make it difficult to track changes. With the release of the stable 5.16 kernel, support will be added to the main kernel for rk356x, and the official assembly of all images using current\edge from sources common to all RK and with a kernel configuration and patches common to all RK will begin. Therefore, it is logical to immediately place all the GENERAL settings in one place. In a separate place, there will be only old legacy versions for 356x (by the way, for this reason, initially, in the first PR, I did it according to the same principles in the legacy rockchip64 group). The whole process of building 356х images uses a single format, exactly the same as all the other aarch64 rk33xx.

Reply:
IMHO separating SoC specific staff into separate families (CPUMAX / BLOB names / BOOT_SCENARIOS) while keeping the generic staff (images format) in the generic rockchip64 family is more readable to new users.
Instead of reading through a hundreds lines long file full of conditionals you only need to consult the short specific ones.

Reply:
I disagree. :) 
A person sees the entire block of functions at once (I pay attention, it is a specific function, for example, setting the processor frequency, setting blob file variables, etc.) it is easier to understand than a bunch of non-explicit overrides in different places. Another nuance. The logic of redefining variables from different points sometimes causes ambiguous behavior and it is much more difficult to find an error if something does not work correctly. I agree that the definition of variables placed in the settings file of a particular board (board.config) is a good solution, even if some of the data will be duplicated in these files. But separating the service data files that are used for all devices of the RK group (the difference is only in the value of the variable) is a bad idea. A separate file is justified if you need to add (or redefine) something that is unique to this group and will interfere with the rest if it is placed in a shared file. The golden rule of a programmer is that the fewer lines of code and fewer points where the function changes, the fewer errors and incorrect operation. ^).

Reply:
> I disagree. :)

As is your right to do so :)
I believe we have a different perception of what `rk35xx.conf` is versus `rockchip64_include.inc`.

For me it is the entry point to the `rk35xx` family (which is well defined and referenced in the board configuration file) and should be the first thing that newcomer dealing with some board in this family sees.
It is there that he/she encounters the configuration belonging only to this narrow family and does not need to care about other members of a broader rockchip64 family.

If one wants to dive in into the image building process it is there in the general helper script "hidden away" and should most probably be SoC agnostic. At least most of it is right now - except for the configurations.

Keep in mind that these might be (probably are) simply personal preferences / perception ;p

Reply:
There is only one correct position - mine .... :) (joke). 

Show at least one variable that is placed in a separate settings file for 356x, which is not defined (used) in the main rockchip64 file ? I really don't understand why to breed entities, where there is no need for it? I repeat, 356x is exactly the same platform (and with the transition to the main core and u-boot, all patches, configurations and sources will be shared) as 33xx, so there is no reason to allocate it to a separate group and spend time tracking the same changes in both..

Reply:
Variables like CPUMAX are indeed already defined in a common rockchip64_include.inc file but it's mostly because there is no better place to do it.
In every board file it would be too much duplication and there is no specific (SoC related place) to put them in.

With rk35xx it's a bit different - it was created because no current legacy kernels support those SoCs and I believe that "polymorphic like" approach is much better than a bunch of conditionals in a common file ;-) Having a separate family according to this approach begs to keep its specific configuration within it.
This approach is not new to Armbian - have a look at sunxi64 family: sunxi50iw6.conf and sunxi50iw9.conf (and others) configure specific family variables in a specific place.

Anyway, while I am not convinced about current solution presented in the PR it does not break any functionality.
Let's get it in :)